libexec/bats-preprocess

```diff
@@ -31,16 +31,26 @@ encode_name() {
 
 tests=()
 index=0
+run_test=1
 pattern='^ *@test  *([^ ].*)  *\{ *(.*)$'
 
 while IFS= read -r line; do
   let index+=1
   if [[ "$line" =~ $pattern ]]; then
     quoted_name="${BASH_REMATCH[1]}"
+    if [ -n "$BATS_TEST_PATTERN" ]; then
+      if [[ "${quoted_name//\"/}" =~ $BATS_TEST_PATTERN ]]; then
```

Comment:
I chasnged mine to `if [[ "${BATS_TEST_PATTERN}" == *"${quoted_name//\"/}"* ]]; then` which allows

BATS_TEST_PATTERN to be a list of tests instead of pattern matching.  (then i renamed the Variables cause its a list not a pattern.) but I am just noting here in case oithers find this excellent solution for a current feature gap.